adapters/bmtm/brightmountainmedia.go

```diff
@@ -0,0 +1,153 @@
+package bmtm
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/mxmCherry/openrtb/v15/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type adapter struct {
+	endpoint string
+}
+
+// Builder builds a new instance of the BrightMountainMedia adapter for the given bidder with the given config.
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter) (adapters.Bidder, error) {
+	bidder := &adapter{
+		endpoint: config.Endpoint,
+	}
+	return bidder, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var extRequests []*adapters.RequestData
+	var errs []error
+
+	for _, imp := range request.Imp {
+		extRequest, err := a.makeRequest(*request, imp)
+		if err != nil {
+			errs = append(errs, err)
+		} else {
+			extRequests = append(extRequests, extRequest)
+		}
+	}
+	return extRequests, errs
+}
+
+func (a *adapter) makeRequest(ortbRequest openrtb2.BidRequest, ortbImp openrtb2.Imp) (*adapters.RequestData, error) {
+	if ortbImp.Banner == nil && ortbImp.Video == nil {
+		return nil, &errortypes.BadInput{
+			Message: fmt.Sprintf("For Imp ID %s Banner or Video is undefined", ortbImp.ID),
+		}
+	}
+
+	var bidderExt adapters.ExtImpBidder
+	if err := json.Unmarshal(ortbImp.Ext, &bidderExt); err != nil {
+		return nil, &errortypes.BadInput{
+			Message: "Error unmarshalling ExtImpBidder",
```

Comment:
Please append the original error to the error message as well. You can do so as follows:
```
Message: fmt.Sprintf("Error unmarshalling ExtImpBidder: %s", err.Error()),
```